controllers/test_utils.go

```diff
@@ -0,0 +1,25 @@
+package controllers
```

Comment:
I would probably move it to `test/` folder and make it a `test` package. This way we can keep test-infra code in one place and exclude it from production code when building the binary.